src/shared/rich-text.js

```diff
@@ -300,7 +307,15 @@ export class RichText extends String {
          */
         await api.trigger('beforeMessageBodyTransformed', this, { 'Synchronous': true });
 
-        this.render_styling && this.addStyling();
+        var text = this.toString();
+
+        if(!this.checkIfHttpsUrl(text)){
+            this.render_styling && this.addStyling();
```

Comment:
Ok I see what you mean. 
Do you happen to know where `this.model.get('references')` is being populated? 
I ask because `this.mentions` is populated by `this.model.get('references')` as seen in `src/shared/chat/message-body.js line 48` but I'm not finding where the `references` field is being populated...

If I can find out how `references` are being identified, I can make a similar implementation for HTTP Urls

Reply:
For incoming messages, the references are extracted when the stanza is parsed:
https://github.com/conversejs/converse.js/blob/a228cf244cccbe3c780a1dbf4b56257c9595ea9f/src/headless/plugins/muc/parsers.js#L225
For outgoing messages it's done somewhere else, I'm not sure right now. However, I don't think this is relevant to fixing this bug.

The `getMediaURLsMetadata` function does the work for you of getting the URLs and you can call that function inside `addStyling` in order to get a list of URLs to ignore.